### ISSUE TITLE
[FluxStoreMV - 1.7.7] Stores are not getting listed

### ISSUE THREAD

**Issue by raziq10000** [inspireui/support#4787]
I have downloaded the latest version of the Flux Store Multi-Vendor app and in that the stores are not getting listed. Please check the logs below.

I/flutter (15751): [2020-08-29 10:07:47.290039Z][Fluxstore][wocommerce_api][15:37:47.290006] getAsync START [endPoint:flutter/wcfm-stores?&page=1] url:http://wclovers.mstore.io/wp-json/wc/v2/flutter/wcfm-stores?oauth_consumer_key=ck_cf6b2a2966f2d908cc08ebaef1336b15f9a6697c&oauth_nonce=nxlqohtoop&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1598695667&oauth_token=&oauth_version=1.0&page=1&oauth_signature=pmJd7U6%2FLf49EPFrlqyeHdVCKR8%3D
I/flutter (15751): [2020-08-29 10:07:49.925904Z][Fluxstore][wocommerce_api][15:37:49.923749] getAsync END [endPoint:flutter/wcfm-stores?&page=1] url:flutter/wcfm-stores?&page=1
E/flutter (15751): [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: Exception: Invalid signature - provided signature does not match.
E/flutter (15751): #0      WCFMApi.searchStores (package:fluxstore_vendor/services/wcfm.dart:188:9)
E/flutter (15751): <asynchronous suspension>
E/flutter (15751): #1      Services.searchStores (package:fluxstore_vendor/services/index.dart:476:23)
E/flutter (15751): #2      StoreModel.loadStore (package:fluxstore_vendor/vendor/store/store_model.dart:87:39)
E/flutter (15751): #3      _StoresState.afterFirstLayout (package:fluxstore_vendor/vendor/store/store_screen.dart:32:16)
E/flutter (15751): #4      AfterLayoutMixin.initState.<anonymous closure> (package:after_layout/after_layout.dart:10:38)
E/flutter (15751): #5      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
E/flutter (15751): #6      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1064:9)
E/flutter (15751): #7      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:972:5)
E/flutter (15751): #8      _rootRun (dart:async/zone.dart:1190:13)
E/flutter (15751): #9      _CustomZone.run (dart:async/zone.dart:1093:19)
E/flutter (15751): #10     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
E/flutter (15751): #11     _invoke (dart:ui/hooks.dart:253:10)
E/flutter (15751): #12     _drawFrame (dart:ui/hooks.dart:211:3)

I have also noticed that the above URL for listing stores with default server configs have params related to oauth. But when I change the URL to point to my domain, the params are not related oauth. Please check the URL below.

https://mydomain/wp-json/wc/v2/flutter/wcfm-stores?&page=1&consumer_key=ck_my_consumer_key&consumer_secret=cs_my_consumer_secret

In both cases, the stores are not getting listed in the Stores tab. Please help me to fix this issue.

**Comment by minhcasi** [comment]
@raziq10000 - pls give a try to test the Connection with your site by using the Fluxbuilder tool - http://fluxbuilder.com

**Comment by raziq10000** [comment]
@minhcasi , I have tested the connection with my site in Flux builder. I am able to search for a product and visit the product details screen. The categories also are getting listed. But the stores screen is just showing me the loader endlessly. I am also not able to visit the store details screen from a product details screen.

**Comment by minhcasi** [comment]
@raziq10000 - could you help to share your config.dart file, we will help to check the issue (you could send privately to inspireui.team@gmail.com) - thank you!

@tgloc96 - pls help with checking this issue

**Comment by raziq10000** [comment]
@minhcasi , @tgloc96 , I have sent the config.dart file to inspireui.team@gmail.com.

**Comment by tgloc96** [comment]
Hello @raziq10000, I checked the feature and everything is working fine.
<img width="531" alt="image" src="https://user-images.githubusercontent.com/46434718/91701025-e5bec180-eba0-11ea-8c1d-7ba8efb7aac8.png">

**Comment by raziq10000** [comment]
@tgloc96 , are you running the version 1.7.7?

**Comment by bizzworld2030** [comment]
Getting the same problem with store listing. All other features are working except store listings.

**Comment by tgloc96** [comment]
> @tgloc96 , are you running the version 1.7.7(https://codecanyon.net/item/fluxstore-multi-vendor-flutter-ecommerce-full-app/25013325)?

Yes, Can you please send me your full source code to tgloc96@gmail.com, I will investigate the issue.

**Comment by tgloc96** [comment]
> Getting the same problem with store listing. All other features are working except store listings.


Can you please send me your full source code to tgloc96@gmail.com, I will investigate the issue.

**Comment by raziq10000** [comment]
> > @tgloc96 , are you running the version 1.7.7(https://codecanyon.net/item/fluxstore-multi-vendor-flutter-ecommerce-full-app/25013325)?
> 
> Yes, Can you please send me your full source code to [tgloc96@gmail.com](mailto:tgloc96@gmail.com), I will investigate the issue.
> 
> > <img alt="" width="48" src="https://camo.githubusercontent.com/3940a8e6f45749988282992ae23b645c0eee12c9/68747470733a2f2f73332e656e7661746f2e636f6d2f66696c65732f3239343635343233312f76656e646f722e706e67">CodeCanyon**[Fluxstore Multi Vendor - Flutter E-commerce Full App](https://codecanyon.net/item/fluxstore-multi-vendor-flutter-ecommerce-full-app/25013325)**
> > 
> > * Version: 1.7.7 – Changelog – Youtube
> > * Last update: 14/08/2020
> > * Files Included: Full Source Code of Flutter App (iOS & Android)
> > * Drag-drop FluxBuilder tool
> > * This is Exclusive Product Li...

I have sent you the full source code and config.dart file. Please have a look.

**Comment by bizzworld2030** [comment]
I have also sent you the full source code which I believe is same as @raziq10000 the issue is synonymous.

**Comment by bizzworld2030** [comment]
> I have also sent you the full source code which I believe is same as @raziq10000 the issue is synonymous.



> > Getting the same problem with store listing. All other features are working except store listings.
> 
> Can you please send me your full source code to [tgloc96@gmail.com](mailto:tgloc96@gmail.com), I will investigate the issue.

I have also sent you the full source code which I believe is same as @raziq10000 the issue is synonymous.

**Comment by tgloc96** [comment]
@raziq10000, I checked your source code and I see the url in the config file you don't change to your website url and key data, it is still my demo website.
<img width="480" alt="image" src="https://user-images.githubusercontent.com/46434718/91792161-135a4800-ec3f-11ea-9076-8e013559b7ab.png">

**Comment by tgloc96** [comment]
> > I have also sent you the full source code which I believe is same as @raziq10000 the issue is synonymous.
> 
> > > Getting the same problem with store listing. All other features are working except store listings.
> > 
> > 
> > Can you please send me your full source code to [tgloc96@gmail.com](mailto:tgloc96@gmail.com), I will investigate the issue.
> 
> I have also sent you the full source code which I believe is same as @raziq10000 the issue is synonymous.

I don't see your source code, please send me again. Can you screen record or screen shot when the error occurred?

**Comment by raziq10000** [comment]
> @raziq10000, I checked your source code and I see the url in the config file don't change to your website url and key data, it is still my demo website.
> <img alt="image" width="480" src="https://user-images.githubusercontent.com/46434718/91792161-135a4800-ec3f-11ea-9076-8e013559b7ab.png">

@tgloc96 , looks like some confusion there. Please note that the source code I sent is having the config.dart file with your demo website URL and key data. I have sent the config.dart file with my website URL and key data separately and I have mentioned the same in the email. Please change the config.dart file to the one in the attachment and test.

**Comment by tgloc96** [comment]
I tested again your config and at the current release version contain some issues it is causing problems with loading store information. For fix this Please download and extract a file below, after that override this file at `lib/vendor/store/store_model.dart`
[store_model.dart.zip](https://github.com/inspireui/support/files/5154831/store_model.dart.zip)

**Comment by kennethhuang31** [comment]
Hi, I think I encounter a similar problem as shown in the pictures. Could u help me to have a look? Thank you.
<img width="346" alt="QQ20200901-204326@2x" src="https://user-images.githubusercontent.com/42427519/91828460-45e26000-ec94-11ea-8e99-61eec4aaceb6.png">
![IMG_7483](https://user-images.githubusercontent.com/42427519/91828485-50045e80-ec94-11ea-9e31-295a5379da68.PNG)

**Comment by bizzworld2030** [comment]
> @raziq10000 I tested again your config and at the current release version contain some issues it is causing problems with loading store information. For fix this Please download and extract a file below, after that override this file at `lib/vendor/store/store_model.dart`
> [store_model.dart.zip](https://github.com/inspireui/support/files/5154831/store_model.dart.zip)

This solution works temporarily for a first time click on the store listing. If you click on category then back to store listing, nothing displays. Also if you get back to any other tab and go back to store listing, nothing displays.

**Comment by tgloc96** [comment]
> > @raziq10000 I tested again your config and at the current release version contain some issues it is causing problems with loading store information. For fix this Please download and extract a file below, after that override this file at `lib/vendor/store/store_model.dart`
> > [store_model.dart.zip](https://github.com/inspireui/support/files/5154831/store_model.dart.zip)
> 
> This solution works temporarily for a first time click on the store listing. If you click on category then back to store listing, nothing displays. Also if you get back to any other tab and go back to store listing, nothing displays.

Thank you for your feedback, I will check right now.

**Comment by tgloc96** [comment]
> > @raziq10000 I tested again your config and at the current release version contain some issues it is causing problems with loading store information. For fix this Please download and extract a file below, after that override this file at `lib/vendor/store/store_model.dart`
> > [store_model.dart.zip](https://github.com/inspireui/support/files/5154831/store_model.dart.zip)
> 
> This solution works temporarily for a first time click on the store listing. If you click on category then back to store listing, nothing displays. Also if you get back to any other tab and go back to store listing, nothing displays.

I checked and the store listing is still displayed normally. Please create a new ticket with describing detail your problem and version you are using, ... and we will support.

**Comment by raziq10000** [comment]
@tgloc96 , I have overwritten store_model.dart and tested the app. Stores are getting listed. But the search stores feature is not working. For every search keyword, it is displaying the same list.

**Comment by minhcasi** [comment]
fixed on next release `1.7.8`

@minhtruong315 - please help to test this issue again - thank you!

**Comment by raziq10000** [comment]
Thanks @minhcasi . Please let me know when will the version 1.7.8 be ready for me to test.

**Comment by minhtruong315** [comment]
@raziq10000 please try to reinstall this MStore Api plugin and try again. I fixed search stores
[mstore-api.zip](https://github.com/inspireui/support/files/5187293/mstore-api.zip)

**Comment by minhtruong315** [comment]
Hi, I'm closing this ticket as there is no response after more than 2 days, if you have any questions or reproducing the issue kindly reopen or create a new ticket - thank you!

**Comment by keegho** [comment]
Im a iOS native swift & flutter developer. I have same issue that stores are not being listed in the app. After some investigation and testing the end point on postman I found out that the store listing is returning and empty array. 

That is because the endpoint being printed in the debug screen is `/wp-json/wc/v2/flutter/wcfm-stores?&page=`1 and page 1 has no data. When I tried page 0 it worked fine. So why is the app calling page 1? Plus every time i use the toggle button in the nav bar to change between the categories and stores the page number is being incremented. Why?

Defiantly there is something wrong here in the code. Any help how to fix that. Still couldn't find from where to `getStores(`) function is being called.
